app/addons/documents/mango/mango.api.js

```diff
@@ -115,15 +140,26 @@ export const mangoQueryDocs = (databaseName, queryCode, fetchParams) => {
     credentials: 'include',
     method: 'POST',
     body: JSON.stringify(modifiedQuery)
-  })
-    .then((res) => res.json())
-    .then((json) => {
-      if (json.error) {
-        throw new Error('(' + json.error + ') ' + json.reason);
-      }
-      return {
-        docs: json.docs,
-        docType: Constants.INDEX_RESULTS_DOC_TYPE.MANGO_QUERY
-      };
-    });
+  });
+};
+
+export const mangoQueryDocs = (databaseName, queryCode, fetchParams) => {
+  return supportsExecutionStats(databaseName).then((shouldFetchExecutionStats) => {
+    if (shouldFetchExecutionStats) {
+      queryCode.execution_stats = true;
```

Comment:
Can you add a note here that we can only add the execution_stats header if it is supported otherwise mango throws errors for unsupported json fields